### ISSUE TITLE
Creating an entry in calendar crashes the app

### ISSUE THREAD

**Issue by arturs105** [toggl/mobileapp#4371]
## 💥 Crash Report
The app crashes when creating time entries in the calendar by long pressing and dragging. Though this isn't easy to reproduce - i had to try multiple times for it to crash. However, i did manage to get a crash twice when there were no time entries for the current day, so that might have something to do with it. But then again, i managed to get a crash when there were several time entries 🤷‍♂️ 

Platforms:
* iPhone X simulator on iOS 12.1 with latest changes from `develop`
* iPhone X on iOS 12.1.2 with latest version from App Store.

### 👥 Impact
Has occured 9 times to 7 people in the last 30 days.

### 🛰 Link
https://appcenter.ms/orgs/Toggl/apps/Toggl-iOS/crashes/errors/1345500516u/overview?q=CalendarCollectionViewSource

### 📝  Exception & Stack Trace
`System.ArgumentOutOfRangeException`
```cs
  at System.Collections.Generic.List`1[T].get_Item (System.Int32 index) [0x00009] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.2.1.13/src/Xamarin.iOS/external/corefx/src/Common/src/CoreLib/System/Collections/Generic/List.cs:180 
  at Toggl.Daneel.ViewSources.CalendarCollectionViewSource.updateCalendarItem (Foundation.NSIndexPath indexPath, System.DateTimeOffset startTime, System.Nullable`1[T] duration) [0x0000d] in /Users/arturs/Projects/mobileapp/Toggl.Daneel/ViewSources/CalendarCollectionViewSource.cs:317 
  at Toggl.Daneel.ViewSources.CalendarCollectionViewSource.UpdateItemView (Foundation.NSIndexPath indexPath, System.DateTimeOffset startTime, System.TimeSpan duration) [0x00019] in /Users/arturs/Projects/mobileapp/Toggl.Daneel/ViewSources/CalendarCollectionViewSource.cs:223 
  at Toggl.Daneel.Views.Calendar.CalendarCollectionViewCreateFromSpanHelper.longPressChanged (CoreGraphics.CGPoint point) [0x000de] in /Users/arturs/Projects/mobileapp/Toggl.Daneel/Views/Calendar/CalendarCollectionViewCreateFromSpanHelper.cs:129 
  at Toggl.Daneel.Views.Calendar.CalendarCollectionViewCreateFromSpanHelper.onLongPress (UIKit.UILongPressGestureRecognizer gesture) [0x000a8] in /Users/arturs/Projects/mobileapp/Toggl.Daneel/Views/Calendar/CalendarCollectionViewCreateFromSpanHelper.cs:68 
  at UIKit.UILongPressGestureRecognizer+Callback.Activated (UIKit.UILongPressGestureRecognizer sender) [0x00000] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.2.1.13/src/Xamarin.iOS/UIKit/UIGestureRecognizer.cs:183 
  at (wrapper managed-to-native) UIKit.UIApplication.UIApplicationMain(int,string[],intptr,intptr)
  at UIKit.UIApplication.Main (System.String[] args, System.IntPtr principal, System.IntPtr delegate) [0x00005] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.2.1.13/src/Xamarin.iOS/UIKit/UIApplication.cs:79 
  at UIKit.UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName) [0x0002c] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.2.1.13/src/Xamarin.iOS/UIKit/UIApplication.cs:63 
  at Toggl.Daneel.Application.Main (System.String[] args) [0x00000] in /Users/arturs/Projects/mobileapp/Toggl.Daneel/Startup/Main.cs:8 
```

**Comment by arturs105** [comment]
Not sure if this is the same crash, but i've experienced several crashes, when trying to create a small entry in the calendar. Small being 15 minutes.

I also noticed that, when the entry is 15 minutes i could get the calendar entry to not show the end time label while creating an entry in the calendar by dragging up and down by [little bits](https://pbs.twimg.com/profile_images/890253808944721921/2zgAeJhh_400x400.jpg). That might be related to crashes as well.

**Comment by arturs105** [comment]
Did a little experimenting today and found out the following:
* Crashes are easier to replicate on an actual device (tried on iPhone X), didn't get one crash on a simulator
* This happens on the latest app store version (1.17) and on latest `develop` as well
* Seems like there could be 2 different exceptions - today i managed to get a bunch of `NullReferenceException`s, which are different from the exception mentioned in the issue
* I think this might be some kind of race condition, since the same steps from the same state sometimes lead to a crash and sometimes don't.
* The race condition might be that `CalendarCollectionViewSource.StopEditing()` is called too early. This causes `CalendarCollectionViewSource.editingItemIndexPath` to become null, and that, in turn, leads to `CalendarCollectionViewSource.IndexPathForEditingItem()` return null. That returned null is used in `CalendarCollectionViewLayout.frameForEditingHour(NSIndex)`, which causes a null exception.

**Comment by dshr** [comment]
Closed in #4440
